### ISSUE TITLE
AbstractViewHelper::isValidType misses more simple types like int, float

### ISSUE THREAD

**Issue by simonschaufi** [TYPO3/Fluid#1021]
When I define an argument as int, it actually tries to load a class named "int" which of course doesn't exist here:

https://github.com/TYPO3/Fluid/blob/main/src/Core/ViewHelper/AbstractViewHelper.php#L361

Instead all simple types should be checked manually without checking if `class_exists`.

**Comment by s2b** [comment]
If we would add scalar types here, a lot of templates would break. I don't think that it's a good idea to throw an exception if the user inputs `'1'` instead of `1`.

Can you elaborate what the exact problem is and what you'd like to achieve?

**Comment by simonschaufi** [comment]
The problem is that fluid looks for classes which are simple types which it shouldn't do. `'1'` will be casted to int actually so this shouldn't be a problem.

**Comment by s2b** [comment]
So it's just an implementation problem, not user-facing?

**Comment by simonschaufi** [comment]
yes, internally only

**Comment by lolli42** [comment]
if i understood correctly, the background of this issue was this reported issue/patch: https://review.typo3.org/c/Packages/TYPO3.CMS/+/87424

**Comment by simonschaufi** [comment]
yes, this is where i originally started from and realized the reported issue here.

**Comment by s2b** [comment]
That's what I had in mind as well, but that's why I asked: The issue is that some input values are not cast correctly. For example, if `maxDepth` is defined as `int`, I would expect to get an integer in the ViewHelper. I would not expect to get an exception.

I'd suggest to close this issue and to write a more general issue instead that describes the problem of the Core issue.

**Comment by s2b** [comment]
Nevermind, sorry. :) I'd suggest creating a separate issue for the casting behavior. This one is fine, though I would not consider it high priority in contrast to the type casting.

**Comment by s2b** [comment]
Related: #1035

**Comment by s2b** [comment]
This will be resolved with Fluid 5. Follow-up issue: #1105
